### ISSUE TITLE
Curves Disappear When Zoomed in and/or Panning

### ISSUE THREAD

**Issue by magicbycalvin** [SciTools/cartopy#1610]
### Description

When plotting curves on a map, I am unable to get some of them to show up. I can get them to show up if I zoom out. I can also get them to show up by zooming out a little bit and then panning so that the map is in the upper right hand corner.

Specifically, I am plotting flight trajectories between cities in the continental US. I am using the Lambert Conformal projection to draw everything. The states use the Plate Carree projection before being transformed. The points representing the cities use the Geodetic projection before being transformed. Finally, the flight trajectories use the Google Mercator projection before being transformed. With the geodetic extent set to `[-125, -66.5, 20, 50]`, I can only see one of four trajectories plotted. If I zoom out by setting the geodetic extent to `[-150, -30, 0, 100]` and pan the states to the right side of the plot, the trajectories will slowly show up. It's not until the extent is set to `[-180, 0, -10, 160]` that all the trajectories will show up (it could be sooner, but this was found with a few trial and error attempts).

I will post the minimum code to reproduce, however, the entirety of the example is also available at https://github.com/caslabuiowa/BeBOT/blob/adding_gjk/Examples/AirTrafficControl.py.

Note that the example I am going for is purely for looks and therefore the accuracy of the trajectories is not that important.

Image without curves being shown:
![cartopy_1](https://user-images.githubusercontent.com/17996040/87937496-7b234d80-ca5a-11ea-8ccf-0c1d7fae9ea5.png)

Zooming out slightly:
![cartopy_2](https://user-images.githubusercontent.com/17996040/87937498-7bbbe400-ca5a-11ea-996c-7818a5d6f3bc.png)

The next three images were produced by panning the slightly zoomed out plot:
![cartopy_3](https://user-images.githubusercontent.com/17996040/87937499-7bbbe400-ca5a-11ea-8a64-4c83acc9e51c.png)
![cartopy_4](https://user-images.githubusercontent.com/17996040/87937500-7c547a80-ca5a-11ea-87e0-3037cf45defd.png)
![cartopy_5](https://user-images.githubusercontent.com/17996040/87937503-7c547a80-ca5a-11ea-8e84-af8c687ea9d0.png)

Zoomed out entirely:
![cartopy_6](https://user-images.githubusercontent.com/17996040/87937505-7c547a80-ca5a-11ea-844a-3a4139695299.png)

#### Code to reproduce
```
from cartopy import crs
from cartopy.io.shapereader import natural_earth, Reader
import matplotlib.pyplot as plt
import numpy as np
from scipy.special import binom


def B(n, i, t, t0, tf):
    coef = binom(n, i)
    num = np.power(t - t0, i) * np.power(tf - t, n-i)
    den = (tf - t0)**n

    return coef*(num/den)


def C(n, P, t, t0, tf):
    i = np.arange(n+1)

    coefs = B(n, i, t, t0, tf)

    return (P*coefs).sum(1)


def BernPoly(cpts, t0=0, tf=1, npts=1001):
    dim = cpts.shape[0]
    n = cpts.shape[1] - 1

    curve = np.empty((dim, npts))

    for idx, t in enumerate(np.linspace(t0, tf, npts)):
        curve[:, idx] = C(n, cpts, t, t0, tf)

    return curve


def drawUS(cities):
    """
    Draws a map of the US with cities marked on the map.

    An example of the cities that could be passed in is,

    cities = [
        [-117.1611, 32.7157],   # San Diego
        [-74.006, 40.7128],     # New York
        [-93.2650, 44.9778],    # Minneapolis
        [-122.3321, 47.6062],   # Seattle
        [-80.1918, 25.7617],    # Miami
        [-104.9903, 39.7392]    # Denver
        ]

    Parameters
    ----------
    cities : list
        2D list containing lat/long pairs of coordinates for each city.

    Returns
    -------
    ax : matplotlib.axes
        Axes on which the US and cities are plotted.

    """
    fig = plt.figure()
    ax = fig.add_axes([0, 0, 1, 1], projection=crs.LambertConformal())
    ax.set_extent([-125, -66.5, 20, 50], crs.Geodetic())

    name = 'admin_1_states_provinces_lakes_shp'
    states = natural_earth(resolution='110m', category='cultural', name=name)

    fc = [0.9375, 0.9375, 0.859375]
    ec = 'black'

    for state in Reader(states).geometries():
        ax.add_geometries([state], crs.PlateCarree(), fc=fc, ec=ec)

    for city in cities:
        ax.plot(city[0], city[1], 'ko', markersize=15,
                transform=crs.Geodetic())

    return ax


if __name__ == '__main__':
    cities = [
        [-117.1611, 32.7157],   # San Diego
        [-74.006, 40.7128],     # New York
        [-93.2650, 44.9778],    # Minneapolis
        [-122.3321, 47.6062],   # Seattle
        [-80.1918, 25.7617],    # Miami
        [-104.9903, 39.7392]    # Denver
        ]
    
    cpts = [np.array([[-13042313.99277981, -13034647.95570562, -11472808.89214453, -11630588.394847, -10389878.34590885, -10382212.30883466], [3857628.40815481, 3857628.40815481, 4767363.68440689, 4480498.22185719, 5618027.22736238, 5618027.22736238]]),
     np.array([[ -8238310.235647, -8251208.14526729, -11402173.3117876,  -10223154.52900812, -13605049.17005154, -13617947.07967182],
      [  4970071.57914243,   4970071.57914243,   5559419.07948082,
         5308830.37536152,   6041588.81740925,   6041588.81740925]]),
     np.array([[-10382212.30883466, -10375038.14982427,  -9828515.17758217,
        -9482865.98337671,  -8926910.34179604,  -8926910.34179604],
      [  5618027.22736238,   5618027.22736238,   4607783.08971783,
         3986150.52509097,   2976770.43752787,   2969596.27851748]]),
     np.array([[-13617947.07967182, -13612395.14326606, -11854576.8527368,
       -13443513.69919485, -11693018.67063879, -11687466.73423303],
      [  6041588.81740925,   6041588.81740925,   4950751.56351547,
         5940019.61476929,   4828115.59603243,   4828115.59603243]])]
    
    trajs = [BernPoly(pt) for pt in cpts]
    
    ax = drawUS(cities)
    for i, traj in enumerate(trajs):
        ax.plot(traj[0], traj[1], label=f'Vehicle {i+1}', transform=crs.GOOGLE_MERCATOR)
    ax.legend()
    plt.show()
```

#### Traceback 
N/A

<details>
 <summary>Full environment definition</summary>

### Operating system
Ubuntu 18.04

### Cartopy version
0.17.0

### conda list

```
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main  
alabaster                 0.7.12                     py_0  
argh                      0.26.2                   py38_0  
astroid                   2.3.3                    py38_0  
atomicwrites              1.4.0                      py_0  
attrs                     19.3.0                     py_0  
autopep8                  1.5.3                      py_0  
babel                     2.8.0                      py_0  
backcall                  0.2.0                      py_0  
bleach                    3.1.5                      py_0  
brotlipy                  0.7.0           py38h7b6447c_1000  
ca-certificates           2020.6.24                     0  
cartopy                   0.17.0           py38h3e2e9bd_1  
certifi                   2020.6.20                py38_0  
cffi                      1.14.0           py38he30daa8_1  
chardet                   3.0.4                 py38_1003  
cloudpickle               1.5.0                      py_0  
cryptography              2.9.2            py38h1ba5d50_0  
cycler                    0.10.0                   py38_0  
cython                    0.29.20          py38he6710b0_0  
dbus                      1.13.16              hb2f20db_0  
decorator                 4.4.2                      py_0  
defusedxml                0.6.0                      py_0  
diff-match-patch          20200713                   py_0  
docutils                  0.16                     py38_1  
entrypoints               0.3                      py38_0  
expat                     2.2.9                he6710b0_2  
flake8                    3.8.3                      py_0  
fontconfig                2.13.0               h9420a91_0  
freetype                  2.10.2               h5ab3b9f_0  
future                    0.18.2                   py38_1  
geos                      3.8.0                he6710b0_0  
glib                      2.65.0               h3eb4bd4_0  
gst-plugins-base          1.14.0               hbbd80ab_1  
gstreamer                 1.14.0               hb31296c_0  
icu                       58.2                 he6710b0_3  
idna                      2.10                       py_0  
imagesize                 1.2.0                      py_0  
importlib-metadata        1.7.0                    py38_0  
importlib_metadata        1.7.0                         0  
intervaltree              3.0.2                      py_1  
ipykernel                 5.3.2            py38h5ca1d4c_0  
ipython                   7.16.1           py38h5ca1d4c_0  
ipython_genutils          0.2.0                    py38_0  
isort                     4.3.21                   py38_0  
jedi                      0.15.2                   py38_0  
jeepney                   0.4.3                      py_0  
jinja2                    2.11.2                     py_0  
jpeg                      9b                   h024ee3a_2  
jsonschema                3.2.0                    py38_0  
jupyter_client            6.1.6                      py_0  
jupyter_core              4.6.3                    py38_0  
keyring                   21.2.1                   py38_0  
kiwisolver                1.2.0            py38hfd86e86_0  
krb5                      1.18.2               h173b8e3_0  
lazy-object-proxy         1.5.0            py38h7b6447c_0  
lcms2                     2.11                 h396b838_0  
ld_impl_linux-64          2.33.1               h53a641e_7  
libblas                   3.8.0               17_openblas    conda-forge
libcblas                  3.8.0               17_openblas    conda-forge
libcurl                   7.71.1               h20c2e04_1  
libedit                   3.1.20191231         h14c3975_1  
libffi                    3.3                  he6710b0_2  
libgcc-ng                 9.1.0                hdf63c60_0  
libgfortran-ng            7.3.0                hdf63c60_0  
liblapack                 3.8.0               17_openblas    conda-forge
libopenblas               0.3.10               h5a2b251_0  
libpng                    1.6.37               hbc83047_0  
libsodium                 1.0.18               h7b6447c_0  
libspatialindex           1.9.3                he6710b0_0  
libssh2                   1.9.0                h1ba5d50_1  
libstdcxx-ng              9.1.0                hdf63c60_0  
libtiff                   4.1.0                h2733197_1  
libuuid                   1.0.3                h1bed415_2  
libxcb                    1.14                 h7b6447c_0  
libxml2                   2.9.10               he19cac6_1  
line_profiler             2.1.2            py38h7b6447c_0  
llvmlite                  0.31.0           py38hd408876_0  
lz4-c                     1.9.2                he6710b0_0  
markupsafe                1.1.1            py38h7b6447c_0  
matplotlib                3.2.2                         0    conda-forge
matplotlib-base           3.2.2            py38hef1b27d_0  
mccabe                    0.6.1                    py38_1  
mistune                   0.8.4           py38h7b6447c_1000  
nbconvert                 5.6.1                    py38_0  
nbformat                  5.0.7                      py_0  
ncurses                   6.2                  he6710b0_1  
numba                     0.48.0           py38hb3f55d8_0    conda-forge
numpy                     1.18.5           py38h8854b6b_0    conda-forge
numpydoc                  1.1.0                      py_0  
olefile                   0.46                       py_0  
opengjk-cython-version    0.0.0                    pypi_0    pypi
openssl                   1.1.1g               h7b6447c_0  
owslib                    0.20.0                     py_0  
packaging                 20.4                       py_0  
pandas                    1.0.5            py38hcb8c335_0    conda-forge
pandoc                    2.10                          0  
pandocfilters             1.4.2                    py38_1  
parso                     0.5.2                      py_0  
pathtools                 0.1.2                      py_1  
pcre                      8.44                 he6710b0_0  
pexpect                   4.8.0                    py38_0  
pickleshare               0.7.5                 py38_1000  
pillow                    7.2.0            py38hb39fc2d_0  
pip                       20.1.1                   py38_1  
pluggy                    0.13.1                   py38_0  
proj                      7.0.1                h59a7b90_1  
prompt-toolkit            3.0.5                      py_0  
psutil                    5.7.0            py38h7b6447c_0  
ptyprocess                0.6.0                    py38_0  
pycodestyle               2.6.0                      py_0  
pycparser                 2.20                       py_2  
pydocstyle                5.0.2                      py_0  
pyepsg                    0.4.0                      py_0  
pyflakes                  2.2.0                      py_0  
pygments                  2.6.1                      py_0  
pykdtree                  1.3.1           py38hdd07704_1002  
pylint                    2.4.4                    py38_0  
pyopenssl                 19.1.0                     py_1  
pyparsing                 2.4.7                      py_0  
pyproj                    2.6.1.post1      py38h61f852b_1  
pyqt                      5.9.2            py38h05f1152_4  
pyrsistent                0.16.0           py38h7b6447c_0  
pyshp                     2.1.0                      py_0  
pysocks                   1.7.1                    py38_0  
python                    3.8.3                hcff3b4d_2  
python-dateutil           2.8.1                      py_0  
python-jsonrpc-server     0.3.4                      py_1  
python-language-server    0.31.9                   py38_0  
python_abi                3.8                      1_cp38    conda-forge
pytz                      2020.1                     py_0  
pyxdg                     0.26                       py_0  
pyyaml                    5.3.1            py38h7b6447c_1  
pyzmq                     19.0.1           py38he6710b0_1  
qdarkstyle                2.8.1                      py_0  
qt                        5.9.7                h5867ecd_1  
qtawesome                 0.7.2                      py_0  
qtconsole                 4.7.5                      py_0  
qtpy                      1.9.0                      py_0  
readline                  8.0                  h7b6447c_0  
requests                  2.24.0                     py_0  
rope                      0.17.0                     py_0  
rtree                     0.9.4                    py38_1  
scipy                     1.5.0            py38h18bccfc_0    conda-forge
secretstorage             3.1.2                    py38_0  
setuptools                49.2.0                   py38_0  
shapely                   1.7.0            py38h98ec03d_0  
sip                       4.19.13          py38he6710b0_0  
six                       1.15.0                     py_0  
snowballstemmer           2.0.0                      py_0  
sortedcontainers          2.2.2                      py_0  
sphinx                    3.1.2                      py_0  
sphinxcontrib-applehelp   1.0.2                      py_0  
sphinxcontrib-devhelp     1.0.2                      py_0  
sphinxcontrib-htmlhelp    1.0.3                      py_0  
sphinxcontrib-jsmath      1.0.1                      py_0  
sphinxcontrib-qthelp      1.0.3                      py_0  
sphinxcontrib-serializinghtml 1.1.4                      py_0  
spyder                    4.1.3            py38h32f6830_0    conda-forge
spyder-kernels            1.9.2                    py38_0  
spyder-line-profiler      0.2.1              pyh9f0ad1d_0    spyder-ide
sqlite                    3.32.3               h62c20be_0  
testpath                  0.4.4                      py_0  
tk                        8.6.10               hbc83047_0  
toml                      0.10.1                     py_0  
tornado                   6.0.4            py38h7b6447c_1  
traitlets                 4.3.3                    py38_0  
ujson                     1.35             py38h7b6447c_0  
urllib3                   1.25.9                     py_0  
watchdog                  0.10.3                   py38_0  
wcwidth                   0.2.5                      py_0  
webencodings              0.5.1                    py38_1  
wheel                     0.34.2                   py38_0  
wrapt                     1.12.1           py38h7b6447c_1  
wurlitzer                 2.0.1                    py38_0  
xz                        5.2.5                h7b6447c_0  
yaml                      0.2.5                h7b6447c_0  
yapf                      0.30.0                     py_0  
zeromq                    4.3.2                he6710b0_2  
zipp                      3.1.0                      py_0  
zlib                      1.2.11               h7b6447c_3  
zstd                      1.4.5                h0b5b093_0 
```

### pip list

```
Package                       Version
----------------------------- -------------------
actionlib                     1.12.0
alabaster                     0.7.12
angles                        1.9.12
argh                          0.26.2
astroid                       2.3.3
atomicwrites                  1.4.0
attrs                         19.3.0
autopep8                      1.5.3
Babel                         2.8.0
backcall                      0.2.0
bleach                        3.1.5
bondpy                        1.8.5
brotlipy                      0.7.0
camera-calibration            1.15.0
camera-calibration-parsers    1.11.13
Cartopy                       0.17.0
catkin                        0.7.26
certifi                       2020.6.20
cffi                          1.14.0
chardet                       3.0.4
cloudpickle                   1.5.0
controller-manager            0.18.1
controller-manager-msgs       0.18.1
cryptography                  2.9.2
cv-bridge                     1.13.0
cycler                        0.10.0
Cython                        0.29.20
decorator                     4.4.2
defusedxml                    0.6.0
diagnostic-analysis           1.9.3
diagnostic-common-diagnostics 1.9.3
diagnostic-updater            1.9.3
diff-match-patch              20200713
docutils                      0.16
dynamic-reconfigure           1.6.3
entrypoints                   0.3
flake8                        3.8.3
future                        0.18.2
gazebo-plugins                2.8.7
gazebo-ros                    2.8.7
gencpp                        0.6.5
geneus                        2.2.6
genlisp                       0.4.16
genmsg                        0.5.16
gennodejs                     2.0.1
genpy                         0.6.12
idna                          2.10
image-geometry                1.13.0
imagesize                     1.2.0
importlib-metadata            1.7.0
interactive-markers           1.11.5
intervaltree                  3.0.2
ipykernel                     5.3.2
ipython                       7.16.1
ipython-genutils              0.2.0
isort                         4.3.21
jedi                          0.15.2
jeepney                       0.4.3
Jinja2                        2.11.2
joint-state-publisher         1.12.15
jsonschema                    3.2.0
jupyter-client                6.1.6
jupyter-core                  4.6.3
kdl-parser-py                 1.13.1
keyring                       21.2.1
kiwisolver                    1.2.0
laser-geometry                1.6.5
lazy-object-proxy             1.5.0
line-profiler                 2.1.2
llvmlite                      0.31.0
MarkupSafe                    1.1.1
matplotlib                    3.2.2
mccabe                        0.6.1
message-filters               1.14.6
mistune                       0.8.4
nbconvert                     5.6.1
nbformat                      5.0.7
numba                         0.48.0
numpy                         1.18.5
numpydoc                      1.1.0
olefile                       0.46
openGJK-cython-version        0.0.0
OWSLib                        0.20.0
packaging                     20.4
pandas                        1.0.5
pandocfilters                 1.4.2
parso                         0.5.2
pathtools                     0.1.2
pexpect                       4.8.0
pickleshare                   0.7.5
Pillow                        7.2.0
pip                           20.1.1
pluggy                        0.13.1
prompt-toolkit                3.0.5
psutil                        5.7.0
ptyprocess                    0.6.0
pycodestyle                   2.6.0
pycparser                     2.20
pydocstyle                    5.0.2
pyepsg                        0.4.0
pyflakes                      2.2.0
Pygments                      2.6.1
pykdtree                      1.3.1
pylint                        2.4.4
pyOpenSSL                     19.1.0
pyparsing                     2.4.7
pyproj                        2.6.1.post1
pyrsistent                    0.16.0
pyshp                         2.1.0
PySocks                       1.7.1
python-dateutil               2.8.1
python-jsonrpc-server         0.3.4
python-language-server        0.31.9
python-qt-binding             0.4.3
pytz                          2020.1
pyxdg                         0.26
PyYAML                        5.3.1
pyzmq                         19.0.1
QDarkStyle                    2.8.1
qt-dotgraph                   0.4.1
qt-gui                        0.4.1
qt-gui-cpp                    0.4.1
qt-gui-py-common              0.4.1
QtAwesome                     0.7.2
qtconsole                     4.7.5
QtPy                          1.9.0
requests                      2.24.0
resource-retriever            1.12.6
rope                          0.17.0
rosbag                        1.14.6
rosboost-cfg                  1.14.9
rosclean                      1.14.9
roscreate                     1.14.9
rosgraph                      1.14.6
roslaunch                     1.14.6
roslib                        1.14.9
roslint                       0.11.2
roslz4                        1.14.6
rosmake                       1.14.9
rosmaster                     1.14.6
rosmsg                        1.14.6
rosnode                       1.14.6
rosparam                      1.14.6
rospy                         1.14.6
rosserial-arduino             0.8.0
rosserial-client              0.8.0
rosserial-python              0.8.0
rosservice                    1.14.6
rostest                       1.14.6
rostopic                      1.14.6
rosunit                       1.14.9
roswtf                        1.14.6
rqt-action                    0.4.9
rqt-bag                       0.4.12
rqt-bag-plugins               0.4.12
rqt-console                   0.4.9
rqt-dep                       0.4.9
rqt-graph                     0.4.11
rqt-gui                       0.5.2
rqt-gui-py                    0.5.2
rqt-image-view                0.4.14
rqt-launch                    0.4.8
rqt-logger-level              0.4.8
rqt-moveit                    0.5.7
rqt-msg                       0.4.8
rqt-nav-view                  0.5.7
rqt-plot                      0.4.9
rqt-pose-view                 0.5.8
rqt-publisher                 0.4.8
rqt-py-common                 0.5.2
rqt-py-console                0.4.8
rqt-reconfigure               0.5.3
rqt-robot-dashboard           0.5.7
rqt-robot-monitor             0.5.9
rqt-robot-steering            0.5.10
rqt-runtime-monitor           0.5.7
rqt-rviz                      0.6.0
rqt-service-caller            0.4.8
rqt-shell                     0.4.9
rqt-srv                       0.4.8
rqt-tf-tree                   0.6.0
rqt-top                       0.4.8
rqt-topic                     0.4.11
rqt-web                       0.4.8
Rtree                         0.9.4
rviz                          1.13.13
scipy                         1.5.0
SecretStorage                 3.1.2
sensor-msgs                   1.12.7
setuptools                    49.2.0.post20200714
Shapely                       1.7.0
sip                           4.19.13
six                           1.15.0
smach                         2.0.1
smach-ros                     2.0.1
smclib                        1.8.5
snowballstemmer               2.0.0
sortedcontainers              2.2.2
Sphinx                        3.1.2
sphinxcontrib-applehelp       1.0.2
sphinxcontrib-devhelp         1.0.2
sphinxcontrib-htmlhelp        1.0.3
sphinxcontrib-jsmath          1.0.1
sphinxcontrib-qthelp          1.0.3
sphinxcontrib-serializinghtml 1.1.4
spyder                        4.1.3
spyder-kernels                1.9.2
spyder-line-profiler          0.2.1
testpath                      0.4.4
tf                            1.12.1
tf-conversions                1.12.1
tf2-geometry-msgs             0.6.5
tf2-kdl                       0.6.5
tf2-py                        0.6.5
tf2-ros                       0.6.5
toml                          0.10.1
topic-tools                   1.14.6
tornado                       6.0.4
traitlets                     4.3.3
ujson                         1.35
urdfdom-py                    0.4.3
urllib3                       1.25.9
watchdog                      0.10.3
wcwidth                       0.2.5
webencodings                  0.5.1
wheel                         0.34.2
wrapt                         1.12.1
wurlitzer                     2.0.1
xacro                         1.13.5
yapf                          0.30.0
zipp                          3.1.0
```

</details>

**Comment by greglucas** [comment]
I can't run that example. But, does it help if you set the extent in PlateCarree() transform?
`ax.set_extent([-125, -66.5, 20, 50], crs.PlateCarree())`

**Comment by magicbycalvin** [comment]
I updated my original comment with code that should work now (assuming you've got the correct packages installed).

I did try your suggestion to set the extent as Plate Carree instead of geodetic but unfortunately that did not work for me.

**Comment by dopplershift** [comment]
I can reproduce with the updated code. Not a clue yet why it's failing. The lines show up fine if you use `Mercator`/`GOOGLE_MERCATOR` for the plotting projection.

**Comment by dopplershift** [comment]
Oh even better, if you change `npts` in `BernPoly` to anything less than 1001 🤨 all the curves show up. I think this is another instance of #1442 . It's good to know, though, that the interactive zoom can make them eventually appear.

I guess I know what I'm doing tonight...

**Comment by magicbycalvin** [comment]
Thanks for your help! I will simply make a plot using your suggestions. This definitely is not my area of expertise, but if there is anything else I can do to help, please let me know!

**Comment by dopplershift** [comment]
So the problem is that we're actually triggering an optimization in matplotlib that doesn't work for us. The work around is, oddly enough, to reset `ax.name` to something other than `'rectilinear'`. So this should also be a fine workaround:
```python
ax.name = 'cartopy'
```

I'm going to close this since it is a duplicate of #1442.
